src/plugins/data/public/types.ts

```diff
@@ -47,6 +51,7 @@ export interface DataPublicPluginSetup {
   search: ISearchSetup;
   fieldFormats: FieldFormatsSetup;
   query: QuerySetup;
+  enhance: (enhancements: DataPublicPluginEnhancements) => void;
```

Comment:
should it take in Partial<enhancements> ? will we be providing enhancements from different plugins ? (one plugin providing search enhancement and another one providing index pattern enhancement) or do we want to limit it to single plugin ? 

if so, lets add a check to make sure we are not calling this twice (and throw)

Reply:
also, lets mark this as internal, i would prefer if we keep it under separate key to make it very clear 
`data.__internal.enhance()`

Reply:
another taught: should we rather provide this functions on per service basis ? (`data.search.enhance()`) or even on per item basis: `data.search.setSearchInterceptor()` (which is what we are currently doing)?

Reply:
a. I don't think we want to take in a partial because this is a pattern for a plugin enhancing it's own matching OSS plugin. Not for a registry or so.
b. I agree about making it internal. I think calling the function `data._enhance` should be enough.
c. We do have it in a per service basis (to propagate the enhancements). However, I think that one of the benefits of having an enhance method rather than a `setInterceptor` method, is the explicitness of it.

An additional change here, is that enhancements are managed during the setup phase, while before we enhanced during the start phase (which was giving me trouble in the Background Sessions PR. IN the same PR I also have multiple enhancements, getting actual value from the pattern suggested here.